src/helics/application_api/ValueFederate.cpp

```diff
@@ -138,37 +146,42 @@ void ValueFederate::registerValueInterfacesJson (const std::string &configString
 {
     auto doc = loadJson (configString);
 
-    if (doc.isMember ("publications"))
+    if (doc.isMember ("publications")) // TODO: add info field loading
     {
         auto pubs = doc["publications"];
         for (const auto &pub : pubs)
         {
             auto key = getKey (pub);
 
-            Publication *pubAct = &vfManager->getPublication (key);
-            if (pubAct->isValid ())
+            auto &pubAct = vfManager->getPublication (key);
```

Comment:
My bad, I was going for uniformity, and hadn't had a chance to do any testing. Is this problem not shared by the `vfManager->getSubscription(key)` on line 190? It was simply having two sections of code that "appear" to be equivalent in purpose be implemented two different ways.

Reply:
It isn't because subAct is never assigned to.  I suppose we should make all those consistent since you can either use pointers or just make a new reference when calling the register function.    Just can't reassign a reference.   I should probably make that a little safer since that might easy to do.    Will have to think about that one a little more.